### PR TITLE
Adjusted combined given-when steps in Tags context

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Combined `given-when` steps splitted into seperate functions and `check-success` assertion added on **Given** steps.

## Related Issue
- Part of https://github.com/owncloud/QA/issues/635

## How Has This Been Tested?

- CI
- :robot:

## Checklist:

- [x] Code changes

### CODE REVIEW

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 1]
```diff
@@ -513,159 +719,413 @@ private function sendProppatchToSystemtags(
...
+	 *
+	 * @param string $user
+	 * @param string $oldName
+	 * @param string $newName
+	 *
+	 * @return void
+	 * @throws \Exception
+	 */
+	public function userHasEditedTheTagWithNameAndSetItsNameToUsingTheWebDAVAPI($user, $oldName, $newName) {
+		$this->editTagName(
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]: can we check for success?

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 2]
```diff
@@ -513,159 +719,413 @@ private function sendProppatchToSystemtags(
...
 	 * @param string $oldName
 	 * @param string $newName
 	 *
 	 * @return void
 	 * @throws \Exception
 	 */
-	public function editTagName($user, $oldName, $newName) {
-		$this->sendProppatchToSystemtags($user, $oldName, 'display-name', $newName);
+	public function theAdministratorHasEditedTheTagWithNameAndSetItsNameToUsingTheWebDAVAPI($oldName, $newName) {
+		$this->editTagWithNameAndSetNameUsingWebDAVAPIAsAdmin(
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]: can we check for success?

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 3]
```diff
@@ -513,159 +719,413 @@ private function sendProppatchToSystemtags(
...
+	 * @Given the user has edited the tag with name :oldName and set its groups to :groups
+	 *
+	 * @param string $oldName
+	 * @param string $groups
+	 *
+	 * @return void
+	 * @throws \Exception
+	 */
+	public function theUserHasEditedTheTagWithNameAndSetsItsGroupsToUsingTheWebDAVAPI($oldName, $groups) {
+		$this->editTagWithNameAndSetGroupsUsingWebDAVAPIAsCurrentUser(
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]: can we check for success?

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 4]
```diff
@@ -513,159 +719,413 @@ private function sendProppatchToSystemtags(
...
+		$this->deleteTagAsCurrentUser($name);
+		$this->featureContext->theHTTPStatusCodeShouldBeSuccess();
+	}
+
+	/**
+	 * @param string $name
+	 *
+	 * @return void
+	 */
+	public function deletesTagAsAdmin($name) {
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]:
```suggestion
	public function deleteTagAsAdmin($name) {
```

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 5]
```diff
@@ -513,159 +719,413 @@ private function sendProppatchToSystemtags(
...
 
+	/**
+	 * @When the administrator deletes the tag with name :name using the WebDAV API
+	 *
+	 * @param string $name
+	 *
+	 * @return void
+	 */
+	public function theAdministratorDeletesTagWithName($name) {
+		$this->deletesTagAsAdmin($name);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]:
```suggestion
		$this->deleteTagAsAdmin($name);
```

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 6]
```diff
@@ -513,159 +719,413 @@ private function sendProppatchToSystemtags(
...
+
+	/**
+	 * @Given the administrator has deleted the tag with name :name
+	 *
+	 * @param string $name
+	 *
+	 * @return void
+	 */
+	public function theAdministratorHasDeletedTagWithName($name) {
+		$this->deletesTagAsAdmin($name);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]:
```suggestion
		$this->deleteTagAsAdmin($name);
```

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 7]
```diff
@@ -727,44 +1187,159 @@ private function requestTagsForFile($user, $fileName, $sharingUser = null) {
...
 	 *
 	 * @return void
+	 * @throws Exception
 	 */
-	public function addsTagToFileFolder(
+	public function userHasAddedTagToFileFolder(
 		$taggingUser, $tagName, $fileName
 	) {
-		$this->tag($taggingUser, $tagName, $fileName);
+		$this->addTagToFileFolder(
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]: check for success

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 8]
```diff
@@ -993,15 +1646,33 @@ public function removesTagFromFile($user, $tagName, $fileName) {
 	 *
 	 * @return void
 	 */
-	public function removesTagFromFileSharedBy(
+	public function userHasRemovedTagFromFileSharedBy(
 		$user, $tagName, $fileName, $shareUser
 	) {
-		$this->untag($user, $tagName, $fileName, $shareUser);
+		$this->removeTagFromFileSharedByUser(
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]: check for success

File: tests/acceptance/features/bootstrap/TagsContext.php [hunk 9]
```diff
@@ -1012,8 +1683,30 @@ public function removesTagFromFileSharedBy(
...
+	 * @param string $tagName
+	 * @param string $fileName
+	 * @param string $shareUser
+	 *
+	 * @return void
+	 */
+	public function theAdministratorHasRemovedTheTagFromFileSharedByUsingTheWebdavApi(
+		$tagName, $fileName, $shareUser
+	) {
+		$this->removeTagFromFileSharedByUserAsAdminUsingWebDavApi(
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TagsContext.php]: check for success

**Review by phil-davis — COMMENTED** [review]
There are some `Given` steps that do not have a check for success - can we check those?

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 9 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/TagsContext.php]: done!

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 8 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/TagsContext.php]: done!

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 7 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/TagsContext.php]: done!

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 6 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/TagsContext.php]: done!

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 5 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/TagsContext.php]: done!

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 4 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/TagsContext.php]: done!

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 3 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/TagsContext.php]: yes..
done!

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 2 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/TagsContext.php]: yes..done!

File: tests/acceptance/features/bootstrap/TagsContext.php (hunk 1 above)
Comment (kiranparajuli589) [reply tests/acceptance/features/bootstrap/TagsContext.php]: yess..done!

**Review by phil-davis — APPROVED** [review]

**Pull request merged**